src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/BuildingRequestResolver.java

```diff
@@ -112,50 +93,39 @@ public void resolve(
         manager.updateRequestState(request.getToken(), RequestState.COMPLETED);
     }
 
-    @NotNull
-    private AbstractBuilding getBuildingFromRequest(
-      @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request) throws RuntimeException
+    @Nullable
+    @Override
+    public IRequest<?> getFollowupRequestForCompletion(
+                                                     @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> completedRequest)
     {
-        final BuildingBasedRequester requester = (BuildingBasedRequester) request.getRequester();
-        return requester.getBuilding().map(r -> {
-            if (r instanceof AbstractBuildingView && manager.getColony() instanceof Colony)
-            {
-                final Colony colony = (Colony) manager.getColony();
-                return colony.getBuildingManager().getBuilding(((AbstractBuildingView) r).getID());
-            }
-            else
-            {
-                return (AbstractBuilding) r;
-            }
-        }).orElseThrow(() -> new IllegalStateException("Unknown building."));
+        return null;
     }
 
     @Nullable
     @Override
-    public IRequest getFollowupRequestForCompletion(
-                                                     @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> completedRequest)
+    public IRequest<?> onRequestCancelled(
+      @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
     {
         return null;
     }
 
     @Nullable
     @Override
-    public IRequest onRequestCancelledOrOverruled(
-                                                   @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request) throws IllegalArgumentException
+    public void onRequestBeingOverruled(
+      @NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
     {
-        return null;
+        return;
     }
 
-    @NotNull
+
     @Override
-    public void onRequestComplete(@NotNull final IToken token)
+    public void onRequestComplete(@NotNull final IRequestManager manager, @NotNull final IToken<?> token)
     {
 
     }
 
-    @NotNull
     @Override
-    public void onRequestCancelled(@NotNull final IToken token)
+    public void onRequestCancelled(@NotNull final IRequestManager manager,@NotNull final IToken<?> token)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)